src/eddymotion/model/optimizers.py

```diff
@@ -0,0 +1,201 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2024 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY kIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+import numpy as np
+from scipy.optimize import minimize
+
+
+def negative_log_likelihood(beta, y, X, kernel, reg_param=1e-6):
+    """
+    Log marginal likelihood function with regularization for any kernel.
+
+    Parameters
+    ----------
+    beta : array-like of shape (n_params,)
+        Log-transformed hyperparameters.
+    y : array-like of shape (n_samples,)
+        Observed data.
+    X : array-like of shape (n_samples, n_features)
+        Input data for kernel function.
+    kernel : callable
+        Kernel function which takes hyperparameters and returns a kernel instance.
+    reg_param : float, default=1e-6
```

Comment:
:obj:`float`

And I would remove the default value. It is shown in the method signature, and by adding it here we run the risk of forgetting and getting it out of sync if the value ever changes in the method signature.

Reply:
Applies also to all methods below.